pkg/executor/chained_task_handle_test.go

```diff
@@ -0,0 +1,193 @@
+package executor
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const taskExecutionTime = 100 * time.Millisecond
+const waitTimeout = 60 * time.Second
+
+func TestSuccsessfulChainedTaskHandle(t *testing.T) {
+	Convey("When having ChainedTaskHandle with successful execution", t, func() {
+		initialTaskHandle := new(MockTaskHandle)
+		initialTaskHandle.On("Wait", 0*time.Second).After(taskExecutionTime).Return(true, nil)
+		initialTaskHandle.On("Stop").Return(nil)
+
+		chainedTaskHandle := new(MockTaskHandle)
+		chainedTaskHandle.On("Wait", 0*time.Second).After(2*taskExecutionTime).Return(true, nil)
+		chainedTaskHandle.On("Stop").Return(nil)
+
+		chainedLauncher := new(MockLauncher)
+		chainedLauncher.On("Launch").Return(chainedTaskHandle, nil)
+
+		Convey("Successful run should yield no error", func() {
+			startTime := time.Now()
+			taskHandle := NewChainedTaskHandle(initialTaskHandle, chainedLauncher)
+			So(taskHandle.Status(), ShouldEqual, RUNNING)
+
+			isTerminated, err := taskHandle.Wait(waitTimeout)
+			So(time.Since(startTime), ShouldBeGreaterThanOrEqualTo, taskExecutionTime)
```

Comment:
This a tad harder than it looks. Unfortunately the in the `initialTaskHandle.On("Wait", 0*time.Second).After(taskExecutionTime).Return(true, nil)` statement, the timer starts *when* the mock is defined - not when the `Wait` function is called. Counter-intuitively, this test runs for `2*taskExecutionTime`.

These problems led to situation, where I could not use `2*taskExecutionTime`, because this might lead to 
```
Expected '199.944883ms' to be greater than '200ms' (but it wasn't)!        
```
So for best effort I have choose to check for at least single execution time. I guess the only thing that could test this case is AssertExpectations.